src/TraceParserGen/TraceParserGen.cs

```diff
@@ -276,8 +304,10 @@ private void GenerateTemplateDefs(TextWriter output)
                 state = ", state";
             }
 
-            output.WriteLine("            return new {0}(action, {1}, {2}, \"{3}\", Guid.Empty, {4}, \"{5}\", ProviderGuid, ProviderName {6});",
-                templateClassName, evnt.Id, evnt.Task, TraceParserGen.ToCSharpName(evnt.TaskName), evnt.Opcode, TraceParserGen.ToCSharpName(evnt.OpcodeName), state);
+            var taskGuid = string.IsNullOrEmpty(evnt.TaskName) ? "Guid.Empty" : ToCSharpName(evnt.TaskName) + "TaskGuid";
```

Comment:
Shouldn't it check m_taskGuids?
If there is no "eventGUID" attributes in parsed manifest, I think it should return Guid.Empty.